app/frontend/src/features/profile/view/leaveReference/LeaveReferencePage.tsx

```diff
@@ -35,29 +36,44 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export default function LeaveReferencePage() {
-  const classes = useStyles(Boolean);
+  const classes = useStyles();
   const { referenceType, userId, hostRequest } = useParams<{
     referenceType: string;
     userId: string;
     hostRequest?: string;
   }>();
 
-  const { data: user, error } = useUser(+userId);
-  const { data: availableRefrences } = useListAvailableReferences(+userId);
+  const { data: user, isLoading: isUserLoading, error: userError } = useUser(
+    +userId
+  );
+  const {
+    data: availableRefrences,
+    isLoading: isAvailableReferencesLoading,
+    error: availableReferencesError,
+  } = useListAvailableReferences(+userId);
 
   if (!(referenceType in ReferenceTypeStrings)) {
     return <Alert severity="error">{INVALID_REFERENCE_TYPE}</Alert>;
   }
 
-  if (availableRefrences && user) {
-    if (referenceType === "friend") {
-      if (
-        availableRefrences.canWriteFriendReference &&
-        user.friends === User.FriendshipStatus.FRIENDS
-      ) {
-        return (
+  return (
+    <>
+      {(userError || availableReferencesError) && (
+        <Alert severity="error">
+          {userError || availableReferencesError?.message || ""}
+        </Alert>
+      )}
+      {(isUserLoading || isAvailableReferencesLoading) && <CircularProgress />}
+
+      {availableRefrences && user ? (
+        (referenceType === referenceTypeRoute[0] &&
```

Comment:
```suggestion
        (referenceType === referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND] &&
```
Ditto for others. This way you don't have to look at the other file